rmgpy/molecule/molecule.py

```diff
@@ -1155,6 +1155,19 @@ def contains_surface_site(self):
                 return True
         return False
 
+    def number_of_surface_sites(self):
+        """
+        Returns the number of surface sites in the molecule.
+        e.g. 2 for a bidentate adsorbate
+        """
+        cython.declare(atom=Atom)
+        cython.declare(count=cython.int)
+        count = 0
+        for atom in self.atoms:
+            if atom.symbol == 'X':
```

Comment:
The class `atom` implements a `is_surface_site()` function that we should use. It does the same thing that you have here (avoiding code duplication) but also if we ever add other symbols for sites we will be more future-proof:
```suggestion
            if atom.is_surface_site():
```